src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -0,0 +1,232 @@
+    InitCode initCode =
+        generator.generateRequestFieldInitCode(context.getMethod(), initFieldStructure, fields);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public InitCodeView generateRequestObjectInitCode(MethodTransformerContext context) {
+    Map<String, Object> initFieldStructure = createInitFieldStructure(context);
+    InitCodeGenerator generator = new InitCodeGenerator();
+    InitCode initCode =
+        generator.generateRequestObjectInitCode(context.getMethod(), initFieldStructure);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public Map<String, Object> createInitFieldStructure(MethodTransformerContext context) {
+    Map<String, String> fieldNamePatterns = context.getMethodConfig().getFieldNamePatterns();
+
+    ImmutableMap.Builder<String, InitValueConfig> initValueConfigMap = ImmutableMap.builder();
+    for (Map.Entry<String, String> fieldNamePattern : fieldNamePatterns.entrySet()) {
+      CollectionConfig collectionConfig = context.getCollectionConfig(fieldNamePattern.getValue());
+      String apiWrapperClassName =
+          context.getNamer().getApiWrapperClassName(context.getInterface());
+      InitValueConfig initValueConfig =
+          InitValueConfig.create(apiWrapperClassName, collectionConfig);
+      initValueConfigMap.put(fieldNamePattern.getKey(), initValueConfig);
+    }
+    Map<String, Object> initFieldStructure =
+        FieldStructureParser.parseFields(
+            context.getMethodConfig().getSampleCodeInitFields(), initValueConfigMap.build());
+    return initFieldStructure;
+  }
+
+  public List<InitCodeLineView> generateSurfaceInitCodeLines(
+      MethodTransformerContext context, InitCode initCode) {
+    List<InitCodeLineView> surfaceLines = new ArrayList<>();
+    for (InitCodeLine line : initCode.getLines()) {
+      switch (line.getLineType()) {
+        case StructureInitLine:
+          surfaceLines.add(generateStructureInitCodeLine(context, (StructureInitCodeLine) line));
+          continue;
+        case ListInitLine:
+          surfaceLines.add(generateListInitCodeLine(context, (ListInitCodeLine) line));
+          continue;
+        case SimpleInitLine:
+          surfaceLines.add(generateSimpleInitCodeLine(context, (SimpleInitCodeLine) line));
+          continue;
+        case MapInitLine:
+          surfaceLines.add(generateMapInitCodeLine(context, (MapInitCodeLine) line));
+          continue;
+        default:
+          throw new RuntimeException("unhandled line type: " + line.getLineType());
+      }
+    }
+    return surfaceLines;
+  }
+
+  public StructureInitCodeLineView generateStructureInitCodeLine(
```

Comment:
Is it possible to have something polymorphism like:

public InitCodeLineView generateInitCodeLineView(MethodTransformerContext context, XXXCodeLine line) {
    ....
}

Then in your generateSurfaceInitCodeLines methond, you just need to call

surfaceLines.add(generateInitCodeLineView(context, line))

so you can just avoid all these switch cases.